### PR TITLE
NOTIF-824 Record metrics when RHOSE returns a server error

### PR DESCRIPTION
**Pull request by gwenneg**

### CODE REVIEW

File: common/src/main/java/com/redhat/cloud/notifications/openbridge/BridgeHelper.java [hunk 1]
```diff
@@ -73,8 +78,12 @@ public Bridge getBridgeIfNeeded() {
         try {
             bridgeList = apiService.getBridgeByName(ourBridgeName, token);
         } catch (WebApplicationException e) {
+            String path = "GET " + BASE_PATH + "?name";
+            rhoseErrorMetricsRecorder.record(path, e);
             if (e.getResponse().getStatus() == 404) {
                 Log.errorf("Bridge with name %s not found in the OpenBridge instance. Did you create it?", ourBridgeName);
+            } else {
+                throw e;
```
Comment (gwenneg) [inline common/src/main/java/com/redhat/cloud/notifications/openbridge/BridgeHelper.java:86]: This introduce a change of the app behavior. The exception was swallowed before, which I think wasn't a good thing.

File: backend/src/test/java/com/redhat/cloud/notifications/rhose/RhoseErrorMetricsRecorderTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        rhoseErrorMetricsRecorder.record("bar", new ServiceUnavailableException());
+        rhoseErrorMetricsRecorder.record("bar", new InternalServerErrorException());
+
+        micrometerAssertionHelper.assertCounterIncrement(COUNTER_NAME, 2L, "path", "foo", "statusCode", "500");
+        micrometerAssertionHelper.assertCounterIncrement(COUNTER_NAME, 1L, "path", "foo", "statusCode", "503");
+
+        micrometerAssertionHelper.assertCounterIncrement(COUNTER_NAME, 1L, "path", "bar", "statusCode", "500");
+        micrometerAssertionHelper.assertCounterIncrement(COUNTER_NAME, 1L, "path", "bar", "statusCode", "503");
+    }
```
Comment (josejulio) [inline backend/src/test/java/com/redhat/cloud/notifications/rhose/RhoseErrorMetricsRecorderTest.java:46]: Could you test that 400's are not counted?

File: common/src/main/java/com/redhat/cloud/notifications/openbridge/RhoseErrorMetricsRecorder.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        try {
+            if (e.getResponse() != null) {
+                int statusCode = e.getResponse().getStatus();
+                if (Family.familyOf(statusCode) == Family.SERVER_ERROR) {
+                    Map<Integer, Counter> pathCounters = counters.computeIfAbsent(path, k -> new HashMap<>());
+                    pathCounters.computeIfAbsent(statusCode, buildCounter(path, statusCode)).increment();
+                }
+            }
+        } catch (Exception ex) {
+            Log.warn("RHOSE error metric recording failed", ex);
```
Comment (josejulio) [inline common/src/main/java/com/redhat/cloud/notifications/openbridge/RhoseErrorMetricsRecorder.java:36]: Would it make sense  to add/log information about the original `path` and `WebApplicationException` that failed to save?

Comment (gwenneg) [reply common/src/main/java/com/redhat/cloud/notifications/openbridge/RhoseErrorMetricsRecorder.java:36]: I don't think this entry will ever be logged. I only added that line to make sure this method would never disturb the methods where it invoked.

File: backend/src/test/java/com/redhat/cloud/notifications/rhose/RhoseErrorMetricsRecorderTest.java (hunk 2 above)
Comment (gwenneg) [reply backend/src/test/java/com/redhat/cloud/notifications/rhose/RhoseErrorMetricsRecorderTest.java:52]: Done.

**Review by josejulio — APPROVED** [review]
lgtm

**Comment by gwenneg** [comment]
The PR check failures are unrelated.

**Pull request merged**